### PR TITLE
MES-4120 - Updated Test Result Provider & Tests

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Added Cat D to the Test Result Provider
- Updated Fault Count Provide to deal with new competencies
- Removed bypass buttons to Pass/Fail a test

## Checklist

- [x] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by szabikr — APPROVED** [review]

File: src/providers/test-result/__tests__/test-result.spec.ts [hunk 1]
```diff
@@ -43,335 +36,67 @@ describe('TestResultCalculatorProvider', () => {
...
-  describe('calculateTestResult', () => {
-    describe('Category B', () => {
-      it('should return a Fail when a dangerous fault exists', (done) => {
-        const testData: CatBUniqueTypes.TestData = {
-          dangerousFaults: {
-            positioningNormalDriving: true,
-          },
-        };
-        testResultProvider.calculateTestResult(TestCategory.B, testData).subscribe((result) => {
+  fdescribe('calculateTestResult', () => {
```
Comment (edwardwilson) [inline src/providers/test-result/__tests__/test-result.spec.ts]: fdescribe needs to be removed

File: src/providers/test-result/test-result.ts [hunk 2]
```diff
@@ -38,6 +42,14 @@ export class TestResultProvider {
...
       case TestCategory.EUAMM1:
         return this.calculateCatAAndSubCategoryTestResult(TestCategory.EUAM1, testData as TestData);
+      case TestCategory.D:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D, testData  as CatDUniqueTypes.TestData);
+      case TestCategory.D1:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D1, testData  as CatD1UniqueTypes.TestData);
+      case TestCategory.DE:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.DE, testData  as CatDEUniqueTypes.TestData);
+      case TestCategory.D1E:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D1E, testData as CatD1EUniqueTypes.TestData);
```
Comment (edwardwilson) [inline src/providers/test-result/test-result.ts]: These can be handled together; would make the switch a bit cleaner

**Review by edwardwilson — CHANGES_REQUESTED** [review]

File: src/providers/test-result/test-result.ts (hunk 2 above)
Comment (TomBillingtonUK) [reply src/providers/test-result/test-result.ts]: fixed  - nice spot

File: src/providers/test-result/test-result.ts [hunk 3]
```diff
@@ -26,18 +30,20 @@ export class TestResultProvider {
...
       case TestCategory.EUAM1:
       case TestCategory.EUA1M1:
       case TestCategory.EUA2M1:
       case TestCategory.EUAMM1:
         return this.calculateCatAAndSubCategoryTestResult(TestCategory.EUAM1, testData as TestData);
+      case TestCategory.D:
+      case TestCategory.D1:
+      case TestCategory.DE:
+      case TestCategory.D1E:
+        return this.calculateCatCDndSubCategoryTestResult(category, testData);
```
Comment (rossfauldsbjss) [inline src/providers/test-result/test-result.ts]: should be named calculateCatDAndSubCategoryTestResult

File: src/providers/test-result/test-result.ts [hunk 4]
```diff
@@ -120,4 +126,27 @@ export class TestResultProvider {
 
     return of(ActivityCodes.PASS);
   }
+
+  private calculateCatCDndSubCategoryTestResult = (
```
Comment (rossfauldsbjss) [inline src/providers/test-result/test-result.ts]: should be named calculateCatDAndSubCategoryTestResult

**Review by rossfauldsbjss — APPROVED** [review]
just a rename of a method due to typo, other than that looks good.

File: src/providers/test-result/test-result.ts (hunk 4 above)
Comment (TomBillingtonUK) [reply src/providers/test-result/test-result.ts]: fixed

File: src/providers/test-result/test-result.ts (hunk 3 above)
Comment (TomBillingtonUK) [reply src/providers/test-result/test-result.ts]: fixed

**Review by edwardwilson — APPROVED** [review]

**Pull request merged**